src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioSymbolNavigationService.cs

```diff
@@ -121,16 +124,15 @@ public bool TryNavigateToSymbol(ISymbol symbol, Project project, OptionSet optio
             // Check whether decompilation is supported for the project. We currently only support this for C# projects.
             if (project.LanguageServices.GetService<IDecompiledSourceService>() != null)
             {
+                var eulaService = project.Solution.Workspace.Services.GetRequiredService<IDecompilerEulaService>();
                 allowDecompilation = project.Solution.Workspace.Options.GetOption(FeatureOnOffOptions.NavigateToDecompiledSources) && !symbol.IsFromSource();
-                if (allowDecompilation && !project.Solution.Workspace.Options.GetOption(FeatureOnOffOptions.AcceptedDecompilerDisclaimer))
+                if (allowDecompilation && !ThreadingContext.JoinableTaskFactory.Run(() => eulaService.IsAcceptedAsync(cancellationToken)))
```

Comment:
why is the eula checking here, instead of where the option is defined in the property pages?

Reply:
The EULA acceptance persists per solution instance.